proctor-common/src/main/java/com/indeed/proctor/common/TestRangeSelector.java

```diff
@@ -156,6 +153,23 @@ private int findMatchingRuleInternal(
         return -1;
     }
 
+    private boolean isValidAllocation(final String identifier, final String rule) {
+        return testDefinition.getTestType().equals(TestType.RANDOM)
+                || isNormalAllocation(identifier)
+                || isUnitlessAllocation(rule);
+    }
+
+    private boolean isNormalAllocation(final String identifier) {
+        return identifier != null
```

Comment:
Should we use StringUtils.hasText(identifier)? It will check null and empty.
So we don't have to implement class NoEmpty implements IdentifierValidator {